postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/PostgreSQLConnection.kt

```diff
@@ -273,20 +286,42 @@ class PostgreSQLConnection @JvmOverloads constructor(
     }
 
     override fun onAuthenticationResponse(message: AuthenticationMessage) {
-
         when (message) {
-            is AuthenticationOkMessage
-            -> {
+            is AuthenticationOkMessage -> {
                 logger.debug("Successfully logged in to database")
                 this.authenticated = true
             }
-            is AuthenticationChallengeCleartextMessage
-            -> {
-                write(this.credential(message))
+            is AuthenticationCleartextPasswordMessage -> write(this.createPasswordMessage(message))
+            is AuthenticationMD5PasswordMessage -> write(this.createPasswordMessage(message))
+            is AuthenticationSASLMessage -> {
+                // Configuration mirrors PostgreSQL's JDBC driver:
+                // https://github.com/ahachete/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jre8/sasl/ScramAuthenticator.java
```

Comment:
As this comment indicates, I leaned heavily on the JDBC implementation for reference. The PR for that change is here, if you'd like to review it as well: https://github.com/pgjdbc/pgjdbc/pull/842/files